coffeecard/CoffeeCard.Models/DataTransferObjects/v2/Product/AddProductRequestWithUserGroups.cs

```diff
@@ -0,0 +1,10 @@
+using System.Collections.Generic;
+using CoffeeCard.Models.Entities;
+
+namespace CoffeeCard.Models.DataTransferObjects.v2.Product;
+
+public class AddProductRequestWithUserGroups
```

Comment:
I think this wrapper class is maybe too much, would it not be simple to have a list of EligibleUserGroups in the AddProductRequest since we only use it in this request anyway